base/sort.jl

```diff
@@ -1298,6 +1298,60 @@ function sort!(A::AbstractArray;
     end
     A
 end
+"""
+    sortperm(A; dims::Integer, alg::Algorithm=DEFAULT_UNSTABLE, lt=isless, by=identity, rev::Bool=false, order::Ordering=Forward)
+
+Return a permutation vector `I` that puts `v[I]` in sorted order along the given dimension. The order is specified
+using the same keywords as [`sort!`](@ref). The permutation is guaranteed to be stable even
+if the sorting algorithm is unstable, meaning that indices of equal elements appear in
+ascending order.
+
+To sort slices of an array, refer to [`sortslices`](@ref).
+
+# Examples
+```jldoctest
+julia> A = [8 7; 5 6]
+2×2 Matrix{Int64}:
+ 8  7
+ 5  6
+
+julia> sortperm(A, dims = 1)
+2×2 Matrix{Int64}:
+ 2  4
+ 1  3
+
+julia> sortperm(A, dims = 2)
+2×2 Matrix{Int64}:
+ 3  1
+ 2  4
+```
+"""
+function sortperm(A::AbstractArray; dims::Integer,
+                  alg::Algorithm=DEFAULT_UNSTABLE,
+                  lt=isless,
+                  by=identity,
+                  rev::Union{Bool,Nothing}=nothing,
+                  order::Ordering=Forward)
+    dim=dims
```

Comment:
This variable is unused, It's up to you if you want to rename `dims` to `dim`, but if you do, use `dim` later.